src/App.tsx

```diff
@@ -1,24 +1,299 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import {
+  FC,
+  useCallback,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
+import toast, { Toaster } from 'react-hot-toast';
 import { UserWarning } from './UserWarning';
 
-const USER_ID = 0;
+import { useLoadingTodosContext } from './contexts/useLoadingTodosContext';
+
+import { TodoFilter } from './components/TodoFilter';
+import { TodoList } from './components/TodoList/TodoList';
+import { NewTodoForm } from './components/NewTodoForm';
+import { TodoItem } from './components/TodoItem';
+import {
+  getTodos,
+  postTodo,
+  deleteTodo,
+  updateTodo,
+} from './api/todos';
+import type { Todo } from './types/Todo';
+import { FilterType } from './enums/FilterType';
+
+enum NotificationType {
+  Success = 'success',
+  Error = 'error',
+  Loading = 'Loading',
+}
+
+const USER_ID = 6712;
+
+const notify = (message: string, type: NotificationType) => {
+  const position = 'top-center';
+  const iconTheme = {
+    primary: '#FFA62B',
+    secondary: '#0F1108',
+  };
+
+  switch (type) {
+    case NotificationType.Error:
+      toast.error(message, {
+        duration: 3000,
+        position,
+        iconTheme,
+      });
+      break;
+    case NotificationType.Success:
+      toast.success(message, {
+        duration: 1000,
+        position,
+        iconTheme,
+      });
+      break;
+    case NotificationType.Loading:
+      toast.loading(message, {
+        duration: 500,
+        position,
+      });
+      break;
+    default:
+      toast(message);
+  }
+};
+
+export const App: FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [filterType, setFilterType] = useState(FilterType.All);
+  const [isInputDisabled, setIsInputDisabled] = useState(false);
+
+  const { setLoadingTodosIds } = useLoadingTodosContext();
+
+  const completedTodos = useMemo(
+    () => todos.filter(({ completed }) => completed),
+    [todos],
+  );
+  const uncompletedTodos = useMemo(
+    () => todos.filter(({ completed }) => !completed),
+    [todos],
+  );
+
+  const [activeTodosCount, completedTodosCount] = useMemo(
+    () => [uncompletedTodos.length, completedTodos.length],
+    [todos],
+  );
```

Comment:
Overkill for useMemo, no need to use it if you read array length, it won't iterate it to get length